### PR TITLE
Issue 2895 - AgentAutoUpgrade: Packaging update for agent auto upgrade

### PR DESCRIPTION
**Pull request by linggao**
Signed-off-by: linggao <linggao@us.ibm.com>

### CODE REVIEW

File: agent-install/agent-auto-upgrade.sh [hunk 1]
```diff
@@ -551,6 +551,13 @@ function get_upgrade_types() {
 # get the directory that this script is located
 script_dir=$(cd "$(dirname "$0")" &> /dev/null && pwd)
 
+# sets the PATH environmental variable
+if [[ $OSTYPE == darwin* ]]; then
+    export PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/Users/Shared/horizon-cli/bin 
+else
+    export PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/horizon/bin
```
Comment (Polber) [inline agent-install/agent-auto-upgrade.sh:558]: Same here

File: anax-in-container/horizon-container [hunk 2]
```diff
@@ -154,20 +154,21 @@ copyHostOSInfo() {
...
+    docker cp $tmpfile $DOCKER_NAME:/etc/host-os-release
 	if [[ $? -ne 0 ]]; then
 		echo "Could not copy host os info into agent docker container"
 		exit 2
 	fi
 
 	# cleanup
-    rm host-os-release
+    rm $tmpfile
 }
```
Comment (Polber) [inline anax-in-container/horizon-container:172]: Looks like there is some inconsistent indentation in this section

File: anax-in-container/horizon-container [hunk 3]
```diff
@@ -404,6 +414,31 @@ restart() {
 	start "$1" 'updating'
 }
 
+add_horizon_upgrade_cronjob() {
+	local container_number=$1
+	echo "Adding agent upgrade cron job..."
+	if isMacos; then
+		echo "A dialog box may be shown to ask your permisson, please click the OK button."
```
Comment (Polber) [inline anax-in-container/horizon-container:421]: permisson -> permission

File: anax-in-container/horizon-container [hunk 4]
```diff
@@ -404,6 +414,31 @@ restart() {
...
+		cronjobs=$(sudo crontab -l 2>/dev/null; echo "* * * * * /usr/local/bin/agent-auto-upgrade.sh ${container_number} 2>&1|/usr/bin/logger -t AgentAutoUpgrade-${container_number}")
+		echo "$cronjobs" | sudo crontab -
+	else
+		sudo echo "* * * * * root /usr/horizon/bin/agent-auto-upgrade.sh ${container_number} 2>&1|/usr/bin/logger -t AgentAutoUpgrade-${container_number}" > /etc/cron.d/horizon${container_number}_agent_upgrade
```
Comment (Polber) [inline anax-in-container/horizon-container:425]: Do we want this running every minute?

File: anax-in-container/horizon-container [hunk 5]
```diff
@@ -404,6 +414,31 @@ restart() {
...
+	fi
+}
+
+remove_horizon_upgrade_cronjob() {
+	local container_number=$1
+	echo "Removing agent upgrade cron job..."
+	if isMacos; then
+		echo "A dialog box may be shown to ask your permisson, please click the OK button."
```
Comment (Polber) [inline anax-in-container/horizon-container:433]: permisson -> permission

**Review by Polber — CHANGES_REQUESTED** [review]

File: agent-install/agent-auto-upgrade.sh (hunk 1 above)
Comment (linggao) [reply agent-install/agent-auto-upgrade.sh:558]: see above

File: anax-in-container/horizon-container (hunk 2 above)
Comment (linggao) [reply anax-in-container/horizon-container:172]: fixed

File: anax-in-container/horizon-container (hunk 4 above)
Comment (linggao) [reply anax-in-container/horizon-container:425]: Changed to every 5 minutes

File: anax-in-container/horizon-container (hunk 5 above)
Comment (linggao) [reply anax-in-container/horizon-container:433]: Fixed

File: pkg/rpm/horizon/horizon.spec [hunk 6]
```diff
@@ -73,10 +73,17 @@ systemctl start horizon.service
 #fi
 mkdir -p /var/horizon/ /var/run/horizon/
 
+# add cron job for agent auto-upgrade
+echo "* * * * * root /usr/horizon/bin/agent-auto-upgrade.sh 2>&1|/usr/bin/logger -t AgentAutoUpgrade" > /etc/cron.d/horizon_agent_upgrade
```
[the remark this answers is not in the document]
Comment (linggao) [reply pkg/rpm/horizon/horizon.spec:77]: Changed to 5 minutes

**Pull request merged**